docs/sources/architecture/distributor.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Distributor
+weight: 1000
+---
+# Distributor Component
+
+This document builds upon the information in the [Loki Architecture](./) page.
+
+## Where does it live?
+
+The distributor is the first component on Loki's write path. It's responsible for validating, preprocessing, and applying a subset of rate limiting to incoming data before sending it to the ingester component. 
+
+## What does it do?
+
+### Validation
+
+The first step the distributor takes is to ensure that all incoming data is according to specification. This includes things like checking that the labels are valid Prometheus labels as well as ensuring the timestamps aren't too old or too new or the log lines aren't too long.
+
+### Preprocessing
+
+Currently the only way the distributor mutates incoming data is by normalizing labels. What this means is making `{foo="bar", bazz="buzz"}` equivalent to `{bazz="buzz", foo="bar"}`, or in other words, ensuring that the order of labels doesn't matter. This allows Loki to cache and hash them deterministically.
```

Comment:
"ensuring that order of labels doesn't matter" is this the same as saying "sorts the labels?" If so I think it would be more clear if it just said "sorts the labels" if not maybe explaining what we are actually doing to make them deterministic is worth adding.